server/instance_cloudoauth.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	jira "github.com/andygrunwald/go-jira"
+	"github.com/mattermost/mattermost-plugin-jira/server/utils"
+	"github.com/mattermost/mattermost-plugin-jira/server/utils/types"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	oauth2_jira "golang.org/x/oauth2/jira"
+)
+
+type cloudOAuthInstance struct {
+	*InstanceCommon
+
+	// The SiteURL may change as we go, so we store the PluginKey when as it was installed
+	MattermostKey string
+}
+
+type OAuth2Token struct {
+	AccessToken string `json:"access_token"`
```

Comment:
Any reason why we're making this struct here instead of using the Golang `oauth2` library's `oauth2.Token` struct here? We'll want access to fields and methods there, e.g. to handle token refreshes